app/services/BulletPointBuilder.scala

```diff
@@ -210,14 +241,35 @@ object BulletPointBuilder {
     *
     * @return - Returns a sub-sample of the input text
     */
-  def extractLeadingMatchedWords(noOfMatchedWords: Int, text: String): String = {
+  def extractLeadingMatchedWords(noOfMatchedWords: Int, texts: List[String], textsProcessed: Int, matches: List[Match], matchesProcessed: Int): String = {
+
+    val textElements: List[Match] = notSpaceRegex.findAllMatchIn(texts(textsProcessed)).toList
```

Comment:
think this is the same without the need to draw on regex and Matches
texts(textsProcessed).split(" +").toList

Reply:
Running the test class the outputs of notSpaceRegex.findAllMatchIn(texts(textsProcessed)).toList and texts(textsProcessed).split(" +").toList are not always equal. In addition using a regex simplifies the extraction of the final element of the leading text. As such I recommend with keep the existing code.